zeppelin-web/src/components/data-validator/data-validator-service.js

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+angular.module('zeppelinWebApp').service('dataValidatorSrv', function($rootScope, mapdataValidator, scatterdataValidator, chartdataValidator, dataModelSchemas) {
+
+  this.validateMapData = function(data) {
```

Comment:
here and below there is a lot of duplicate code.
this duplication must be eliminated.